File: HotelReservationServiceJava/RoomTypeA.java
import java.util.Arrays;
import java.util.stream.Stream;

public class RoomTypeA extends Room
{
    protected double pricePerDay;
    
    public RoomTypeA(int capacity, double pricePerDay){
        super();
        this.capacity = capacity;
        this.pricePerDay = pricePerDay;
    }
    
    //This kind of room only accounts duration to calculate totalCost
    @Override
    public double totalCost(){
        return Arrays.stream(availability).filter(x -> x != null)
                        .mapToDouble(x -> pricePerDay).sum();
    }
}

File: HotelReservationServiceCpp/src/RoomTypeA.cpp
#include "RoomTypeA.h"

RoomTypeA::RoomTypeA(int cap, double ppDay) :
    pricePerDay{ppDay},
    Room(cap, 0)
{}

double RoomTypeA::totalCost(){
    double cost = 0;
    for(int i = 0; i < 30; i++){
        if(availability[i] != nullptr){
            cost += pricePerDay;
        }
    }
    return cost;
}

RoomTypeA::~RoomTypeA(){}
